ros2param/ros2param/api/__init__.py

```diff
@@ -17,20 +17,44 @@
 from rcl_interfaces.srv import GetParameters
 from rcl_interfaces.srv import SetParameters
 import rclpy
+import yaml
 
 
 def get_parameter_value(*, string_value):
     """Guess the desired type of the parameter based on the string value."""
     value = ParameterValue()
-    if string_value.lower() in ('true', 'false'):
+    try:
+        yaml_value = yaml.load(string_value)
+    except yaml.parser.ParserError:
+        value.type = ParameterType.PARAMETER_STRING
+        value.string_value = string_value
+        return value
+
+    if isinstance(yaml_value, bool):
         value.type = ParameterType.PARAMETER_BOOL
-        value.bool_value = string_value.lower() == 'true'
-    elif _is_integer(string_value):
+        value.bool_value = yaml_value
+    elif isinstance(yaml_value, int):
         value.type = ParameterType.PARAMETER_INTEGER
         value.integer_value = int(string_value)
```

Comment:
This could assign `yaml_value` instead which makes the `int()` call is obsolete.

Same for `float` below.